diff --git a/PhysicsTools/NanoAOD/python/custom_jme_cff.py b/PhysicsTools/NanoAOD/python/custom_jme_cff.py
index ca21e78095d..44123012ddc 100644
--- a/PhysicsTools/NanoAOD/python/custom_jme_cff.py
+++ b/PhysicsTools/NanoAOD/python/custom_jme_cff.py
@@ -509,6 +509,7 @@ def SavePatJets(proc, jetName, payload, patJetFinalColl, jetTablePrefix, jetTabl
     )
   )
   getattr(proc,jetTableName).variables.pt.precision=10
+  getattr(proc,jetTableName).variables.rawFactor.precision=10
 
   #
   # Save MC-only jet variables in table
@@ -638,6 +639,8 @@ def ReclusterAK4PuppiJets(proc, recoJA, runOnMC):
     jetPuppiTableDoc += "For jets with pt < 8 GeV, only those matched to AK4 Gen jets are stored."
   proc.jetPuppiTable.doc = jetPuppiTableDoc
 
+  proc.jetPuppiTable.variables.rawFactor.precision = 10
+
   #
   # Add variables
   #
@@ -795,6 +798,8 @@ def ReclusterAK4CHSJets(proc, recoJA, runOnMC):
     jetTableDoc += "For jets with pt < 8 GeV, only those matched to AK4 Gen jets are stored."
   proc.jetTable.doc   = jetTableDoc
 
+  proc.jetTable.variables.rawFactor.precision = 10
+
   #
   # Add variables
   #
@@ -1033,6 +1038,8 @@ def AddVariablesForAK8PuppiJets(proc):
   Add more variables for AK8 PFPUPPI jets
   """
 
+  proc.fatJetTable.variables.rawFactor.precision = 10
+
   #
   #  These variables are not stored for AK8PFPUPPI (slimmedJetsAK8)
   #  in MiniAOD if their pt < 170 GeV. Hence the conditional fill.
